src/AllowListDrop.sol

```diff
@@ -407,6 +431,62 @@ contract AllowListDrop is
         return firstMintedTokenId;
     }
 
+    function gasLessPurchase(
+        uint256 quantity,
+        address recipient,
+        string memory _formResponse
+    )
+        external
+        payable
+        nonReentrant
+        canMintTokens(quantity)
+        onlyPublicSaleActive
+        returns (uint256)
+    {
+        uint256 salePrice = salesConfig.publicSalePrice;
+        address erc20PaymentToken = salesConfig.erc20PaymentToken;
+        address fundsRecipient = config.fundsRecipient;
+
+        if (erc20PaymentToken == address(0)) {
+            if (msg.value != salePrice * quantity) {
+                revert Purchase_WrongPrice(salePrice * quantity);
+            }
+        } else {
+            IERC20Upgradeable(erc20PaymentToken).transferFrom(
+                recipient,
+                fundsRecipient,
+                salePrice * quantity
+            );
+        }
```

Comment:
what is this?

why do we have a new method, just for gasless?

Reply:
it was easier on testing thats why, this was done a while ago but i can refactor if you wish